### DISCUSSION TITLE
How to load third-party extension's JS file?

### DISCUSSION THREAD

**Discussion by help-meetanshi** [breezefront/community#12] [1. Help]
Hi,

In the Breeze theme, when we set "Enable Breeze Experience" as "Use theme settings" or "Yes", our Magento 2 Mobile Login extension's JS file is not loading.

https://d.pr/vLXuwi

If our Magento 2 Mobile Login extension's JS file doesn't load, then our extension cannot work.

Will you please guide us to overcome this situation?

**Comment by vovayatsyuk** [comment]
You need to integrate your js with Breeze. 

I recommend reading our docs: https://breezefront.com/components

Then, check how we integrate our own modules:

**Easytabs**

- Layout update: https://github.com/swissup/module-easytabs/blob/master/view/frontend/layout/breeze_default.xml
- JS: https://github.com/swissup/module-easytabs/blob/master/view/frontend/web/js/breeze/tabs.js
- CSS: https://github.com/swissup/module-easytabs/blob/master/view/frontend/web/css/breeze/_default.less

**Testimonials**

- Layout: https://github.com/swissup/module-testimonials/blob/master/view/frontend/layout/breeze_default.xml
- JS: https://github.com/swissup/module-testimonials/blob/master/view/frontend/web/js/breeze/testimonials.js

**Reply by help-meetanshi** [reply]
Hi,
Thanks for providing the document link.
With the help of your given document, our Magento 2 Mobile Login extension's JS file is loading now.
But we cannot create an object of our extension's JS file as per default Magento's require config method.
Is there any other way to create an object of JS file while using a Breeze theme?
Will you please help us with that?

**Reply by vovayatsyuk** [reply]
Please show me the code you are trying to write. What is the “object of js file”?

**Reply by vovayatsyuk** [reply]
We've just released a new version that simplifies integration steps. Please update and try to use these tips: https://breezefront.com/integration-tips

**Reply by help-meetanshi** [reply]
Hi,
Thanks for providing the new version link.

Please check the following screenshot of our code.
https://drops.meetanshi.com/P7D46m

Code File: https://drops.meetanshi.com/dxaqZE

We want to execute the code written in the require dependency injection.
But we are not able to do so.

Please help us with that.

**Reply by vovayatsyuk** [reply]
You don't need `require` when writing new file. Here is a simple js code:

```
(function () {
    'use strict';

    $(document).on('breeze:load', function () {
        console.log('Hello');
    });
})();
```

**Reply by help-meetanshi** [reply]
Hi,
We have written require because we need the following files and want to use their functionalities.
'jquery',
'mage/url',
'Magento_Ui/js/modal/modal',
'mage/validation'
Please let me know how can we achieve that.

**Reply by vovayatsyuk** [reply]
I think I'd better add a repo example with various code snippets that works in both Luma and Breeze frontends.

If you want to include your use case in that example, please provide a code of how you use these dependencies:

- modal
- validation

**Reply by help-meetanshi** [reply]
Hi,
Please find our extension JS file in the following zip file URL.
https://drops.meetanshi.com/jp9IZg

In that JS file, we are using the following dependencies.
`'jquery',
'mage/url',
'Magento_Ui/js/modal/modal',
'mage/validation'`

**Reply by help-meetanshi** [reply]
Any update for us?

**Reply by vovayatsyuk** [reply]
In breeze you don't need to require dependencies, they all are already loaded. So instead of `modal()` do this:

```js
$('.mobile-sendotp-popup').modal(sendoption);
```

Validation should work without the changes:

```js
$('#otp-login-form').validation('isValid');
```

mage/url was added in latest breeze version:

```js
$.breeze.url.build()

// or

require(['mage/url'], function (URL) {
    url.build();
});
```

And here is your code:

```js
require([
    'jquery',
    'mage/url',
    'Magento_Ui/js/modal/modal',
    'mage/validation'
], function ($, url, modal, validation) {
    'use strict';

    var sendoption = {
        type: 'popup',
        modalClass: 'mobile_login_popup',
        responsive: true,
        innerScroll: true,
        clickableOverlay: false,
        buttons: [],
    };

    $(document).on('breeze:load', () => {
        $('.mobile-sendotp-popup').modal(sendoption);
    });

    $(document).on('click', '.mt_signin', function (e) {
        $('.mobile-sendotp-popup').modal('openModal');
        $('.login-otp-title').click();
    });

    $(document).on('click', '.request-otp-btn', function (e) {
        var validForm = $('#otp-login-form').validation('isValid');
        if (validForm) {
            var mobilenumber = $('#mobilenumber').val();
            var getUrl = url.build('mobilelogin/otp/send');
            var otptype = 'login';
            var resend = 0;

            $.ajax({
                url: getUrl,
                type: 'POST',
                data: {
                    mobilenumber: mobilenumber,
                    otptype: otptype,
                    resendotp: resend,
                    oldmobile: 0
                },
                showLoader: true,
                success: function (response) {
                    if (response.succeess === "true") {
                        $('fieldset.fieldset.login-otp.step1').hide();
                        $('fieldset.fieldset.login-otp.step2').show();
                        $('p.success-msg').text(response.successmsg);
                        $('p.success-msg').show();
                    }
                    else {
                        $('p.err-msg').text(response.errormsg);
                        $('p.err-msg').show();
                    }
                }
            });
        }
    });
});
```

**Reply by help-meetanshi** [reply]
Hi,
Thanks for providing the example code.

1) We are getting the following error with the $('p.success-msg').show().delay(5000).fadeOut(); code.
https://d.pr/VMLkiq
It seems that the delay() is not working.

2) The $.breeze.url.build() code is also not working and gives the following error.
https://d.pr/ewbgj0

Can we have a meeting with a developer to expedite the process?

**Reply by vovayatsyuk** [reply]
1. We do not have delay and animations inside js. We do not have jquery. Please stick to example I posted to you. 
2. As I told in prev comment it was added in latest version. Please update.

**Reply by help-meetanshi** [reply]
Hi,
We already using the latest version.
"swissup/breeze-blank": "1.2.1"

Please let me know if we are missing any other thing.

**Reply by vovayatsyuk** [reply]
Use `composer outdated "swissup/*"` command to see if everything is up to date. For this particular case you need to update swissup/module-breeze to 1.14.4.

Here is a commit where I added 'mage/url' functionality: https://github.com/breezefront/module-breeze/commit/d293620222c2ae95351302fefc459384a5d2fe19

**Reply by help-meetanshi** [reply]
Hi,
Now we can use the $.breeze.url.build() code.
Thanks for your help.

Currently, we are getting the following errors while using some other JS files of default Magento.

1) Code: https://d.pr/DEARz5
Error: https://drops.meetanshi.com/R8efAj

2) Code: https://d.pr/j4TVvT
Error: https://drops.meetanshi.com/Vyaleh

Please help to fix the above issues as well.

Can you also let us know how to add our extension's custom CSS file?

**Reply by vovayatsyuk** [reply]
1. You have to write all of this logic. We do not have replacements for checkout files. 
2. You are calling modal without initialization. In breeze you have to initialize it first. 

Css: See the docs.

**Reply by help-meetanshi** [reply]
Hi,
Now we can load our CSS and JS files.
Thanks for your help.

Please check the following screenshot where you can find the JS files loaded from the Meetanshi_Mobilelogin extension.
https://drops.meetanshi.com/i/RqYuor

But those JS files are not loading on the checkout page.
Please check the following screenshot.
https://drops.meetanshi.com/i/gpvLQw

Do you know reason of it?

Currently, we are loading the JS files with the following file.
https://drops.meetanshi.com/f/OqHpMv

Do we need to write any other code to load JS on the checkout page?

**Reply by help-meetanshi** [reply]
Any update for us?

**Reply by vovayatsyuk** [reply]
Yes, since breeze js is disabled at checkout page you need to write standard “luma” code to include your js files.
